group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroup.java

```diff
@@ -411,6 +432,20 @@ public int numMembers() {
         return members.size();
     }
 
+    /**
+     * @return The number of members that use the legacy protocol.
+     */
+    public int numLegacyProtocolMember() {
+        return (int) members.values().stream().filter(member -> member.useLegacyProtocol()).count();
```

Comment:
It may be better to maintain this count in the group state instead of having to go through all members. Is it possible?